lib/graphql/schema/member/relay_shortcuts.rb

```diff
@@ -31,13 +31,13 @@ def edge_type
           end
         end
 
-        def connection_type
+        def connection_type(node_nullable: true)
```

Comment:
The weird thing about this is: 

- We accept `node_nullable:` as true or false;
- We cache the generated type after the first call; 
- But if a later call includes a _different_ `node_nullable` value, we return the already-generated type (even though it was generated using a different `node_nullable` value. 

I don't think the library should behave like that because it'd be very unexpected behavior to pass one input to a method, but get a return value as though a different input was provided. 

Here's how you could do something like this now: 

```ruby 
# Make a custom base connection class that defaults to non-null nodes
class Types::BaseConnection < GraphQL::Types::Relay::BaseConnection 
  # set the default to false here 
  def self.edge_type(*args, node_nullable: false, **kwargs)
    super 
  end 
end 

# ... 
# Then hook it up to other base classes: 

class Types::BaseObject < ... 
  connection_type_class(Types::BaseConnection)
end 

class Types::BaseUnion < ...
  connection_type_class(Types::BaseConnection)
end 

module Types::BaseInterface 
  # ...
  connection_type_class(Types::BaseConnection)
end 
```

That way, the `.connection_type` helper will always generate a connection with `node_nullable: false`. 

Other options might be: 

- Caching the value of the first `node_nullable:` and raising an error if a connection type is generated with a different value. 
- Caching one generated type _per_ `node_nullable:` value (extending the cache to a Hash of `{ node_nullable_value => type_class }`, or using two instance variables)

Personally, I think those two are more trouble than they're worth. In my experience, adding a base class pays of in the long run. 

What do you think of those approaches?

Reply:
I see. I wasn't aware of the possibility that another input could be made after the type cache was generated.

>Personally, I think those two are more trouble than they're worth.

I agree with you.
I'll close this PR. 
Thank you for your comment!